app/serializers/serialize_exercise_assignment.rb

```diff
@@ -42,16 +42,26 @@ def hints
     #  ## Header
     #  - foo
     #  - bar
-    hints_doc.each_cons(2).each_with_object({}) do |(header, list), hints|
-      # TODO: Add an issue to the relevant track via a async job
-      # if any of these are invalid
-      next unless header.type == :header
-      next unless header.header_level == 2
-      next unless list.type == :list
 
-      task_id = parse_task_id(header)
-      hints[task_id] = list.each.map { |list_item| Markdown::Parse.(list_item.each.first.to_commonmark) }
-    end
+    hints_doc.
+      each.
+      # Create chunks of nodes whenever the node types is a level 2 heading
```

Comment:
As the code is not obvious, I've added a comment to explain its working. Let me know if this comment is clear enough.